client/components/language-picker/index.jsx

```diff
@@ -65,8 +67,9 @@ class LanguagePicker extends PureComponent {
 		} );
 	}
 
-	toggleOpen = () => {
+	onClick = ( event ) => {
```

Comment:
Code all looks great, my only pick up would be on the naming here as [it could be more descriptive](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md#react-components) of it's purpose.
`toggleOpen` could remain as it's a descriptive action and could potentially be used outside of clicks - a key press or as part of some other action for instance.
That said, it doesn't make sense to call a passed handler inside of `toggleOpen` so we would need another method.
Perhaps keeping `toggleOpen` but also adding `onClick` or `onClickHandler` (to avoid confusion) could work, something like:

```js
toggleOpen() {
  this.setState( { open: ! this.state.open } );
}

onClickHandler = ( event ) => {
  if ( ! this.props.disabled ) {
    this.props.onClick( event );
    this.toggleOpen();
  }
}
```
I'm mostly just thinking out loud here and I don't think this would be a blocker - keen to hear your thoughts!